src/channel/mod.rs

```diff
@@ -11,4 +18,101 @@ trait Channel {
     fn random_number(&mut self, bound: u64) -> u64;
 
     fn random_field(&mut self) -> Self::Field;
+
+    /// Only relevant for non-interactive channels. Changes the channel seed to a "safer" seed.
+    /// 
+    /// This function guarantees that randomness fetched from the channel after calling this function
+    /// and before sending data from the prover to the channel, is "safe": A malicious
+    /// prover will have to perform 2^security_bits operations for each attempt to randomize the fetched
+    /// randomness.
+    /// 
+    /// Increases the amount of work a malicious prover needs to perform, in order to fake a proof.
+    #[allow(unused_variables)]
+    fn apply_proof_of_work(&mut self, security_bits: usize) -> Option<()> {
```

Comment:
I was thinking about how best to do this.

If the PoW is at the end of the transcript it might make sense for this to take ownership of the channel: it finishes the transcript.

In any case, it should return a `Result<(), SomeErroType>` :)

Reply:
Alternatively, since it is: 

"Only relevant for non-interactive channels"

You could introduce another trait:

```
trait FSChannel: Channel {
    fn apply_proof_of_work(&mut self, security_bits: usize) -> Option<()>;
}
```

That way we know at the type level which code is FS specific.

Reply:
Yeah, I also think implementing an additional trait for non-interactive protocols is the best approach. 👍

Reply:
Choose the first way, as there are already too many inheritances.
Note that there is no implementation for non-FS channels yet, though